modules/tensor_mechanics/include/utils/RankTwoScalarTools.h

```diff
@@ -418,75 +418,53 @@ stressIntensity(const RankTwoTensorTempl<T> & stress)
 template <typename T>
 T
 getQuantity(const RankTwoTensorTempl<T> & tensor,
-            const MooseEnum scalar_type,
+            const MooseEnum & scalar_type,
             const Point & point1,
             const Point & point2,
             const Point & curr_point,
             Point & direction)
```

Comment:
It's a bit odd why direction isn't constant. The methods above do change the direction. I hope that is a feature not a bug @dschwen ?

Reply:
Yes, direction is an output value